lib/memo_wise.rb

```diff
@@ -351,6 +351,27 @@ def #{method_name}#{args_str}
         klass.send(visibility, method_name)
       end
     end
+
+    unless target.singleton_class?
+      # Create certain class methods to implement .preset_memo_wise
+      %i[
+        preset_memo_wise
+        validate_memo_wised!
+        validate_params!
+        fetch_key
+      ].each do |method_name|
+        # Like calling 'module_function', but original method stays public
+        target.define_singleton_method(
+          method_name,
+          MemoWise.instance_method(method_name)
+        )
+
+        # Make private the class method copies of private instance methods
+        unless MemoWise.public_method_defined?(method_name)
```

Comment:
What's the case where `MemoWise.public_method_defined?(method_name)` would return `true`?

Reply:
Hi Jemma! In the current code, some of these methods are private helpers and would return false. `preset_memo_wise` is the public method and returns true